lib/OrdinaryDiffEqCore/src/solve.jl

```diff
@@ -119,6 +120,10 @@ function DiffEqBase.__init(
         @warn("Dense output is incompatible with saveat. Please use the SavingCallback from the Callback Library to mix the two behaviors.")
     end
 
+    if !(alias isa ODEAliases)
```

Comment:
Should check if the other alias kwargs are used and throw a depwarn in that case.

And we should allow / handle boolean `alias` specification as described in the higher level issue.